sort_sorttype.go

```diff
@@ -143,14 +142,26 @@ func (s *SortTypeSorter) buildChunks() error {
 	defer close(s.chunkChan) // if this is not called on error, causes a deadlock
 
 	for {
-		c := newChunk(s.config.ChunkSize, s.lessFunc)
-		for i := 0; i < s.config.ChunkSize; i++ {
+		c := newChunk(s.lessFunc)
+		var size int
+	Loop:
```

Comment:
I don't like using break labels here. It makes it hard to understand and debug.
Could this be re-written with loops/functions instead?